### ISSUE TITLE
7.4.0 Start-Process with -PassThru & -RedirectStandardOutput does not return the process object completely ( properties like Exitcode )

### ISSUE THREAD

**Issue by derkveenhof** [PowerShell/PowerShell#20716]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

When i use Start-Process with -PassThru & -RedirectStandardOutput , the process object is not returned completely. This has been since 7.4.0

### Expected behavior

```console
This was the 7.3.9 behavior:

PS C:\Users\thiscouldbeme> $Process = start-process notepad -PassThru -wait -RedirectStandardOutput c:\temp\sa.txt
PS C:\Users\thiscouldbeme> $Process | Select *

Name                       :
Id                         : 33776
PriorityClass              :
FileVersion                :
HandleCount                :
WorkingSet                 :
PagedMemorySize            :
PrivateMemorySize          :
VirtualMemorySize          :
TotalProcessorTime         : 00:00:00
SI                         :
Handles                    :
VM                         :
WS                         :
PM                         :
NPM                        :
Path                       :
CommandLine                :
Parent                     : System.Diagnostics.Process (pwsh)
Company                    :
CPU                        : 0
ProductVersion             :
Description                :
Product                    :
__NounName                 : Process
SafeHandle                 : Microsoft.Win32.SafeHandles.SafeProcessHandle
Handle                     : 2500
BasePriority               :
ExitCode                   : 0
HasExited                  : True
StartTime                  : 17-11-2023 18:31:48
ExitTime                   : 17-11-2023 18:31:50
MachineName                : .
MaxWorkingSet              :
MinWorkingSet              :
Modules                    :
NonpagedSystemMemorySize64 :
NonpagedSystemMemorySize   :
PagedMemorySize64          :
PagedSystemMemorySize64    :
PagedSystemMemorySize      :
PeakPagedMemorySize64      :
PeakPagedMemorySize        :
PeakWorkingSet64           :
PeakWorkingSet             :
PeakVirtualMemorySize64    :
PeakVirtualMemorySize      :
PriorityBoostEnabled       :
PrivateMemorySize64        :
ProcessorAffinity          :
SessionId                  :
StartInfo                  :
Threads                    :
VirtualMemorySize64        :
EnableRaisingEvents        : False
StandardInput              :
StandardOutput             :
StandardError              :
WorkingSet64               :
SynchronizingObject        :
MainModule                 :
PrivilegedProcessorTime    : 00:00:00
UserProcessorTime          : 00:00:00
ProcessName                :
MainWindowHandle           : 0
MainWindowTitle            :
Responding                 : True
Site                       :
Container                  :
```

### Actual behavior

```console
PS C:\Windows\System32> $Process = start-process notepad -PassThru -wait -RedirectStandardOutput c:\temp\sa.txt
PS C:\Windows\System32> $Process | Select *

Name                       :
Id                         : 3264
PriorityClass              :
FileVersion                :
HandleCount                :
WorkingSet                 :
PagedMemorySize            :
PrivateMemorySize          :
VirtualMemorySize          :
TotalProcessorTime         :
SI                         :
Handles                    :
VM                         :
WS                         :
PM                         :
NPM                        :
Path                       :
CommandLine                :
Parent                     :
Company                    :
CPU                        :
ProductVersion             :
Description                :
Product                    :
__NounName                 : Process
SafeHandle                 :
Handle                     :
BasePriority               :
ExitCode                   :
HasExited                  : True
StartTime                  :
ExitTime                   :
MachineName                : .
MaxWorkingSet              :
MinWorkingSet              :
Modules                    :
NonpagedSystemMemorySize64 :
NonpagedSystemMemorySize   :
PagedMemorySize64          :
PagedSystemMemorySize64    :
PagedSystemMemorySize      :
PeakPagedMemorySize64      :
PeakPagedMemorySize        :
PeakWorkingSet64           :
PeakWorkingSet             :
PeakVirtualMemorySize64    :
PeakVirtualMemorySize      :
PriorityBoostEnabled       :
PrivateMemorySize64        :
ProcessorAffinity          :
SessionId                  :
StartInfo                  :
Threads                    :
VirtualMemorySize64        :
EnableRaisingEvents        : False
StandardInput              :
StandardOutput             :
StandardError              :
WorkingSet64               :
SynchronizingObject        :
MainModule                 :
PrivilegedProcessorTime    :
UserProcessorTime          :
ProcessName                :
MainWindowHandle           : 0
MainWindowTitle            :
Responding                 : True
Site                       :
Container                  :
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.0
PSEdition                      Core
GitCommitId                    7.4.0
OS                             Microsoft Windows 10.0.20348
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
Good find, but it is a duplicate of:

* #5421 

See also:

* #20400

In light of the recently introduced bot action of closing old posts that haven't had activity in a while, it may be worth keeping this one open, however.

**Comment by derkveenhof** [comment]
Caching the Handle seems to work.

This workaround works:
$Process = Start-Process Notepad -PassThru -RedirectStandardOutput c:\temp\sa.txt
$Handle = $Process.Handle
$Process.WaitForExit()
$Process | Select *

**Comment by mklement0** [comment]
Good to know a workaround. Just to spell it out: it won't work if `-Wait` is used (as in #20400).

**Comment by MarkDordoy** [comment]
Can also confirm I've seen the same issue.

**Comment by astrahan87** [comment]
+1

**Comment by AFAde** [comment]
Same issue here, please provide a fix

**Comment by mklement0** [comment]
A fix _may_ land in  v7.4.1 (the PR mentions only the related #20400 issue):

* #20866
